subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -28,14 +28,21 @@ static int wifi_connect(uint32_t mgmt_request, struct net_if *iface,
 
 	LOG_HEXDUMP_DBG(params->ssid, params->ssid_length, "ssid");
 	LOG_HEXDUMP_DBG(params->psk, params->psk_length, "psk");
+	if (params->sae_password) {
+		LOG_HEXDUMP_DBG(params->sae_password, params->sae_password_length, "sae");
+	}
 	NET_DBG("ch %u sec %u", params->channel, params->security);
 
-	if ((params->security > WIFI_SECURITY_TYPE_PSK) ||
```

Comment:
So it's always up the user/application to request the connection with proper security mode? I mean does it work as follows?
- the application scans for WiFi networks and is provided with a list containing the security mode offered by each AP
- the  application sends connection request with WiFi mgmt API (and connection request data includes the security mode provided by the scan operation)

Does the supplicant (or WiFi driver) verify if the AP really supports the requested security mode? Can the application define some rules which security modes are supported (I guess it can be done simply by filtering scan results security-wise)?

Reply:
Yes, application has to specifically configure wpa_supplicant to use a security method rather than auto-selecting based on AP, and in case of any mismatch either wpa_supplicant or the AP will fail the association. 

> Can the application define some rules which security modes are supported (I guess it can be done simply by filtering scan results security-wise)?

Yes, this is all left to the application and/or user. We can assign a number to strength of security and filter.